statement-table/src/generic.rs

```diff
@@ -187,6 +187,30 @@ pub struct Summary<D, G> {
 	pub signalled_bad: bool,
 }
 
+/// A validity attestation.
+#[derive(Clone, PartialEq, Decode, Encode)]
+pub enum ValidityAttestation<S> {
+	/// implicit validity attestation by issuing.
+	/// This corresponds to issuance of a `Candidate` statement.
+	Implicit(S),
+	/// An explicit attestation. This corresponds to issuance of a
+	/// `Valid` statement.
+	Explicit(S),
+}
+
+/// An attested-to candidate.
+#[derive(Clone, PartialEq, Decode, Encode)]
+pub struct AttestedCandidate<Group, Candidate, AuthorityId, Signature> {
```

Comment:
Are we ever using this differently than with `C::GroupId, C::Candidate, C::AuthorityId, C::Signature`? Because if not, I'd prefer to bind it to `C: Context` and use them in the def, shortening the return signatures below.

Reply:
we might in the future, with BLS aggregation. It's hard to tell where that will land so I prefer to keep this generic for now.